internal/storage/scsi/scsi.go

```diff
@@ -96,22 +94,24 @@ func ControllerLunToName(ctx context.Context, controller, lun uint8) (_ string,
 
 	scsiID := fmt.Sprintf("0:0:%d:%d", controller, lun)
 
-	// Query for the device name up until the timeout.
+	// Devices matching the given SCSI code should each have a subdirectory
+	// under /sys/bus/scsi/devices/<scsiID>/block.
+	blockPath := filepath.Join("/sys/bus/scsi/devices", scsiID, "block")
 	var deviceNames []os.FileInfo
-	startTime := time.Now()
 	for {
-		// Devices matching the given SCSI code should each have a subdirectory
-		// under /sys/bus/scsi/devices/<scsiID>/block.
-		var err error
-		deviceNames, err = ioutil.ReadDir(filepath.Join("/sys/bus/scsi/devices", scsiID, "block"))
+		deviceNames, err = ioutil.ReadDir(blockPath)
```

Comment:
It looks like if this call succeeds we never exit the loop.

Reply:
I have no idea how this passed tests. Good catch. Let me fix and try again.